### PR TITLE
Clean python requirements

### PR DESCRIPTION
**Pull request by tbille**
# Summary

- Remove unused pyhton requirements (stastd, prometheus-flask-exporter, gunicorn[gevent] * 2 (lol) )
- Reorder requirements in alphabetical order

# QA

-` ./run`
- Make sure snapcraft is still running

### CODE REVIEW

**Comment by nottrobin** [comment]
@tbille If you're changing these dependencies, you might as well use the opportunity to update to [`talisker==0.10.2`](https://pypi.org/project/talisker/).

**Comment by nottrobin** [comment]
@tbille (although keep a close eye out - `0.11` might have landed by the time this PR is merged =) )

**Comment by tbille** [comment]
@nottrobin Can't upgrade to `0.10.2`:

```
Traceback (most recent call last):
  File "/usr/local/bin/talisker.gunicorn", line 11, in <module>
    sys.exit(run_gunicorn())
  File "/usr/local/lib/python3.6/dist-packages/talisker/__init__.py", line 239, in run_gunicorn
    import talisker.celery
  File "/usr/local/lib/python3.6/dist-packages/talisker/celery.py", line 35, in <module>
    from raven.contrib.celery import SentryCeleryHandler
  File "/usr/local/lib/python3.6/dist-packages/raven/contrib/celery/__init__.py", line 12, in <module>
    from celery.exceptions import SoftTimeLimitExceeded
ModuleNotFoundError: No module named 'celery'
```

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,12 @@
+black==18.6b4
```
Comment (nottrobin) [inline requirements.txt]: I like the idea, that I copied from some other codebase that I can't remember, of separating our dev dependencies in this file, like [in docs.snapcraft.io](https://github.com/canonical-websites/docs.snapcraft.io/blob/80c96f7658938bc9c9776208b1d3f05f58d32364/requirements.txt):

```
talisker==0.11.0
responses==0.10.4

# Development dependencies
black
coverage
```

**Comment by nottrobin** [comment]
@tbille you only added a "code needed" label, but I would say when you're updating dependencies, it's really one of the most important times to have a QA...

**Comment by nottrobin** [comment]
@tbille :+1: fair enough. Now just merge this already!

**Pull request merged**